server/tests/api_v4/authenticated/test_shamir_recovery_setup.py

```diff
@@ -0,0 +1,409 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import pytest
+
+from parsec._parsec import (
+    DateTime,
+    RevokedUserCertificate,
+    ShamirRecoveryBriefCertificate,
+    ShamirRecoveryShareCertificate,
+    ShamirRevealToken,
+    authenticated_cmds,
+)
+from tests.common import Backend, CoolorgRpcClients
+from tests.common.client import setup_shamir_for_coolorg
+
+
+async def test_authenticated_shamir_recovery_setup_ok(
+    coolorg: CoolorgRpcClients, with_postgresql: bool
+) -> None:
+    if with_postgresql:
+        pytest.skip("TODO: postgre not implemented yet")
+    share = ShamirRecoveryShareCertificate(
+        author=coolorg.alice.device_id,
+        user_id=coolorg.alice.user_id,
+        timestamp=DateTime.now(),
+        recipient=coolorg.mallory.user_id,
+        ciphered_share=b"abc",
+    )
+    brief = ShamirRecoveryBriefCertificate(
+        author=coolorg.alice.device_id,
+        user_id=coolorg.alice.user_id,
+        timestamp=DateTime.now(),
+        threshold=1,
+        per_recipient_shares={coolorg.mallory.user_id: 2},
+    )
+
+    setup = authenticated_cmds.v4.shamir_recovery_setup.ShamirRecoverySetup(
+        b"abc",
+        ShamirRevealToken.new(),
+        brief.dump_and_sign(coolorg.alice.signing_key),
+        [share.dump_and_sign(coolorg.alice.signing_key)],
+    )
+    rep = await coolorg.alice.shamir_recovery_setup(setup)
+    assert rep == authenticated_cmds.v4.shamir_recovery_setup.RepOk()
+
+    # TODO dump
+
+
+async def test_authenticated_shamir_recovery_setup_shamir_setup_already_exists(
+    coolorg: CoolorgRpcClients, with_postgresql: bool
+) -> None:
+    # Setup previous shamir
+    if with_postgresql:
+        pytest.skip("TODO: postgre not implemented yet")
```

Comment:
I'd use `xfail` which usually means "this should fail because something is not implemented _yet_". `skip` is more for permanent failures (e.g. "this would _always_ fail on Windows platform").

Also, note that you can specify a condition with [`skipif(condition, reason)`](https://docs.pytest.org/en/6.2.x/skipping.html#id1) and [`xfail(condition, reason)`](https://docs.pytest.org/en/6.2.x/skipping.html#condition-parameter) 

So:
```suggestion
    pytest.xfail(with_postgresql, "TODO: postgre not implemented yet")
```